android/src/org/kiwix/kiwixmobile/KiwixMobileActivity.java

```diff
@@ -652,97 +650,61 @@ private void newTabInBackground(String url) {
       @Override
       public void onReceivedTitle(WebView view, String sTitle) {
         super.onReceivedTitle(view, sTitle);
-        mLeftArrayAdapter.notifyDataSetChanged();
+        tabDrawerAdapter.notifyDataSetChanged();
       }
     });
     webView.loadUrl(url);
     webView.loadPrefs();
     mWebViews.add(webView);
-    mLeftArrayAdapter.notifyDataSetChanged();
+    tabDrawerAdapter.notifyDataSetChanged();
     setUpWebView();
     htmlUtils.initInterface(webView);
   }
 
-  private KiwixWebView restoreTabAtIndex(String url, int index) {
-    KiwixWebView webView = new KiwixWebView(KiwixMobileActivity.this);
-    webView.setWebViewClient(new KiwixWebViewClient(KiwixMobileActivity.this, mLeftArrayAdapter));
-    webView.setWebChromeClient(new KiwixWebChromeClient());
-    webView.loadUrl(url);
-    webView.loadPrefs();
-
-
-    mWebViews.add(index, webView);
-    mLeftArrayAdapter.notifyDataSetChanged();
-    selectTab(mWebViews.size() - 1);
+  private void restoreTab(int index) {
+    mWebViews.add(index, tempForUndo);
```

Comment:
Not a new problem with this pull but it appears restoring a tab when only one tab is open fails as a double content provider is called e.g content://base...content://base.../url